### PR TITLE
Do not prevent default on keydown as that will prevent the dispatch of the following keypress event.

### PR DESCRIPTION
**Pull request by awtcode**
Add test_keydown_preventdefault_proxy to test_browser.py for automated testing.

### CODE REVIEW

**Comment by awtcode** [comment]
@kripken , this is the follow-up to the previous pull request on https://github.com/kripken/emscripten/pull/3870. I have added an automated test 'test_keydown_preventdefault_proxy' for this fix.

File: src/proxyClient.js [hunk 1]
```diff
@@ -240,7 +240,13 @@ function cloneObject(event) {
 ['keydown', 'keyup', 'keypress', 'blur', 'visibilitychange'].forEach(function(event) {
   document.addEventListener(event, function(event) {
     worker.postMessage({ target: 'document', event: cloneObject(event) });
-    event.preventDefault();
+    
+    // Do not prevent default on keydown as that will prevent the dispatch of
+    // the following keypress event
+    if (event.type != 'keydown')
```
Comment (kripken) [inline src/proxyClient.js]: please use `!==`, and put the `{` on this line

**Comment by kripken** [comment]
I am somewhat concerned that the SDL port does special-casing of backspace and tab, and this doesn't. Is there a reason not to?

Also, let's move towards sharing this code. How about putting the decision part in a method in `src/library_browser.js` on the `Browser` object (maybe `shouldPreventDefault`?). Then we might be able to move the SDL2 code to JS and reuse that, eventually.

**Comment by awtcode** [comment]
@kripken , I didn't add the special casing for backspace and tab because it wasn't affecting my use case but I did some googling and found that this would cause the user to navigate to the previous page when he uses backspace when entering text so I have added this in.

However, after moving the code to src/library_browser, I am not able to access the Browser object in proxyClient.js. Is there an easy way to include the Browser object in proxyClient.js? Thanks.

**Comment by kripken** [comment]
Oh, I see. Yes, then nevermind about moving to Browser, sorry about that.

Ok, is this ready for review?

**Comment by awtcode** [comment]
@kripken , sorry, I believe I erroneously did a rebase and rewrote the history for my incoming branch. Will close this and submit a new pull request.

**Pull request closed without merging**